src/strands/models/bedrock.py

```diff
@@ -443,7 +445,24 @@ def _stream(
                             for event in self._generate_redaction_events():
                                 callback(event)
 
-                    callback(chunk)
+                    # Track if we see tool use events
+                    if "contentBlockStart" in chunk and chunk["contentBlockStart"].get("start", {}).get("toolUse"):
+                        has_tool_use = True
+
+                    # Fix stopReason for streaming responses that contain tool use
+                    if "messageStop" in chunk and has_tool_use:
+                        message_stop = chunk["messageStop"]
+                        if message_stop.get("stopReason") == "end_turn":
+                            # Create corrected chunk with tool_use stopReason
+                            modified_chunk = chunk.copy()
+                            modified_chunk["messageStop"] = message_stop.copy()
+                            modified_chunk["messageStop"]["stopReason"] = "tool_use"
+                            logger.info("Override stop reason from end_turn to tool_use")
```

Comment:
Let's make this a `warn` instead of an `info`.  This is something I want to have be visible by default to callers